tests/zero_code_change/test_tensorflow2_error_handling_agent.py

```diff
@@ -0,0 +1,406 @@
+
+    return HookWithBadKerasAndLayerCallback
+
+
+@pytest.fixture
+def hook_class_with_keras_callback_error_and_custom_debugger_configuration(
+    out_dir, custom_configuration_error_message, hook_class_with_keras_callback_error
+):
+    class HookWithBadKerasCallbackAndCustomDebuggerConfiguration(
+        hook_class_with_keras_callback_error
+    ):
+        """
+        KerasHook subclass that extends off of the TF Keras callback error subclass above to return a hook with a
+        custom debugger configuration. Thus, any errors thrown should not be caught by the error handling agent.
+        """
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(
+                keras_error_message=custom_configuration_error_message, *args, **kwargs
+            )
+
+        @classmethod
+        def create_from_json_file(cls, json_file_path=None):
+            return HookWithBadKerasCallbackAndCustomDebuggerConfiguration(
+                out_dir=out_dir, include_collections=[CollectionKeys.ALL]
+            )
+
+    return HookWithBadKerasCallbackAndCustomDebuggerConfiguration
+
+
+@pytest.fixture
+def profiler_config_path(config_folder):
+    return os.path.join(config_folder, "test_tf2_profiler_config_parser_by_time.json")
+
+
+@pytest.fixture
+def hook_class_with_keras_callback_error_and_custom_profiler_configuration(
+    out_dir,
+    custom_configuration_error_message,
+    monkeypatch,
+    profiler_config_path,
+    hook_class_with_keras_callback_error,
+):
+    class HookWithBadKerasCallbackAndCustomProfilerConfiguration(
+        hook_class_with_keras_callback_error
+    ):
+        """
+        KerasHook subclass that extends off of the TF Keras callback error subclass above to return a hook with a
+        custom profiler configuration. Thus, any errors thrown should not be caught by the error handling agent.
+        """
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(
+                keras_error_message=custom_configuration_error_message, *args, **kwargs
+            )
+
+        @classmethod
+        def create_from_json_file(cls, json_file_path=None):
+            monkeypatch.setenv("SMPROFILER_CONFIG_PATH", profiler_config_path)
+            return HookWithBadKerasCallbackAndCustomProfilerConfiguration(out_dir=out_dir)
+
+    return HookWithBadKerasCallbackAndCustomProfilerConfiguration
+
+
+@pytest.fixture(autouse=True)
+def set_up_logging_and_error_handling_agent(out_dir, stack_trace_filepath):
+    """
+    Set up each test to:
+        - Add a logging handler to write all logs to a file (which will be used to verify caught errors in the tests)
+        - Remove the duplicate logging filter
+        - Reset the error handling agent after the test so that smdebug is reenabled.
+    """
+    old_create_from_json = Hook.create_from_json_file
+    del_hook()
+
+    logger = get_logger()
+    os.makedirs(out_dir)
```

Comment:
You can move these repetitive operations to the fixture

Reply:
These operations are in the fixture?